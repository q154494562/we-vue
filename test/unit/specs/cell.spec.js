import { shallowMount } from '@vue/test-utils'
import Cell from '@/cell'

describe('cell', () => {
  let wrapper
  afterEach(() => {
    wrapper && wrapper.destroy()
  })

  test('create', () => {
    wrapper = shallowMount(Cell, {
      propsData: {}
    })

    expect(wrapper.name()).toBe('wv-cell')
    expect(wrapper.classes()).toContain('weui-cell')

    expect(wrapper).toMatchSnapshot()
  })

  test('is-link', () => {
    wrapper = shallowMount(Cell, {
      propsData: {
        isLink: true
      }
    })

    expect(wrapper.classes()).toContain('weui-cell_access')

    expect(wrapper).toMatchSnapshot()
  })

  test('handle click', () => {
    const routerLinkSpy = jest.fn()
    wrapper = shallowMount(Cell, {
      propsData: {},
      methods: {
        routerLink: routerLinkSpy
      }
    })

    wrapper.trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
    expect(routerLinkSpy).toHaveBeenCalled()

    expect(wrapper).toMatchSnapshot()
  })
})
