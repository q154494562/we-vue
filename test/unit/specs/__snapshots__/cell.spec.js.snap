// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cell create 1`] = `
<div class="weui-cell">
  <div class="weui-cell_hd"></div>
  <div class="weui-cell__bd">
    <p></p>
  </div>
  <div class="weui-cell__ft"></div>
</div>
`;

exports[`cell handle click 1`] = `
<div class="weui-cell">
  <div class="weui-cell_hd"></div>
  <div class="weui-cell__bd">
    <p></p>
  </div>
  <div class="weui-cell__ft"></div>
</div>
`;

exports[`cell is-link 1`] = `
<div class="weui-cell weui-cell_access">
  <div class="weui-cell_hd"></div>
  <div class="weui-cell__bd">
    <p></p>
  </div>
  <div class="weui-cell__ft"></div>
</div>
`;
